### ISSUE TITLE
Spring boot starter disables scheduling

### ISSUE THREAD

**Issue by ScreamingTree86** [javamelody/javamelody#643]
Hi,
i use the springboot starter of javameldoy for my springboot application. At the switch from version 1.67.0 to 1.68.0 some things go wrong for spring scheduling. I use the springboot annotation `@EnableScheduling` so that the springframework can search for `@Scheduling` annotated beans.

In 1.67.0 everything works as expected but since version 1.68.0 of javamelody my scheduling classes are no longer found by the `ScheduledAnnotationBeanPostProcessor`. 
In general it seems like every bean with scheduling annotations can no longer be resolved by spring with java melody 1.68.0

**Comment by jfbreault** [comment]
Seem to be because of this https://github.com/javamelody/javamelody/blob/master/javamelody-spring-boot-starter/src/main/java/net/bull/javamelody/JavaMelodyAutoConfiguration.java#L150

I had issue with `@EnableAspectJAutoProxy(proxyTargetClass=true)` doing nothing. Disabling the javamelody autoconfiguration and add my own(copy paste the original without DefaultAdvisorAutoProxyCreator bean) and everything seem to work. I'll try to do a pull request for it.

**Comment by evernat** [comment]
I agree that there is an issue with `@EnableScheduling` and `@Scheduled` when enabling the javamelody-spring-boot-starter and spring-boot-starter-aop/aspectj in the app. For example, when adding in pom.xml:
```xml
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-aop</artifactId>
		</dependency>
```
But, if the DefaultAdvisorAutoProxyCreator is removed by PR #644 from the javamelody-spring-boot-starter and if the `spring-boot-starter-aop` or `aspectjweaver` dependencies are not present in the app, then the javamelody advisors for monitoring the executions on classes/methods having `@MonitoredWithSpring, @Service, @Controller, @RestController, @Async, @Scheduled` are not working anymore.
That is, the DefaultAdvisorAutoProxyCreator was really needed in the default case (without the `spring-boot-starter-aop` dependency).

So I think that the PR #644 can probably not be merged.

But perhaps we can add on the JavaMelodyAutoConfiguration.defaultAdvisorAutoProxyCreator method:
`@ConditionalOnProperty(prefix = JavaMelodyConfigurationProperties.PREFIX, name = "advisor-auto-proxy-creator-enabled", matchIfMissing = true)`
At least, you would be able to add in your application.yml:
```yml
javamelody:
  init-parameters:
    advisor-auto-proxy-creator-enabled: false
```
At the moment, I have not seen a better solution for this DefaultAdvisorAutoProxyCreator vs spring-aop/aspectj catch-22.
Or you may question the fact that ScheduledAnnotationBeanPostProcessor uses `AopUtils.getTargetClass(bean)` to find `@Scheduled` methods.

**Comment by jfbreault** [comment]
as soon as you are using aspectj wich is used in 100% of my project in there, it seem to break. I would disable it by default. As soon as you are using sprind-data, `@Transactional`, @scheduled it seem to break. Almost all serious springboot application will use aop/aspectj.

**Comment by jfbreault** [comment]
Or maybe, add spring-boot-starter-aop as you dependencies like spring-data does? Let spring create the aop like he want

**Comment by evernat** [comment]
I think that adding spring-aop will cause issues to some apps. spring-aop may not be used in 100% of projects out of your company. So the new #644 may not be merged either.

**Comment by jfbreault** [comment]
I thinks it's causing more issue like this, exemple project can have issue with it but any business size project cannot use the starter. I did'nt heard of any spring project that provide his own AutoProxyCreator. Spring itself will not do it for you if you need it for a module(transactional, schedule, retry,..) since it can affect to much stuff. So they document what it need to be enabled onder the hood for the requested feature.


PS I just found an other issue with SB 1.5.4(should be 1.5.x but didn't tested to older one), datasource genereated by SB will throw exception since the properties validation seem to fail with jdk proxy.

**Comment by chrisco484** [comment]
We were looking for a Java based monitoring tool for our Tomcat servers and we liked Javamelody but it lost points when we discovered it was based on Spring and therefore, apart from the "magic" issues, has the resultant resource and performance issues Spring can bring.
After successfully avoiding Spring since its inception - choosing standard, non obfuscated ways of provisioning dependencies etc., - I was forced to use it for the last 2 years in a contract. I have developed a theory: the amount of time devs spend dealing with Spring issues is usually about 5-10 times the amount of time it would have taken to just write the 2-3 lines of visible, totally obvious, clear and, best of all, debuggable Java code that avoids the need for Spring's obfuscated "magic" in the first place.

**Comment by evernat** [comment]
@chrisco484 
You can be at ease. javamelody is not based on Spring. For example, the monitoring of http and sql requests and the monitoring of plain java services are not based on Spring. Spring configuration with javamelody is just an option among other configurations.

**Comment by evernat** [comment]
The issue is fixed by ddff401
And this is released in 1.68.1 (to be available soon in maven central repo).
So the methods annotated with [@Scheduled](http://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/scheduling/annotation/Scheduled.html) or [@Schedules](http://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/scheduling/annotation/Schedules.html) are no longer monitored automatically.

To enable the monitoring of those methods, see the updated [doc](https://github.com/javamelody/javamelody/wiki/SpringBootStarter).
Thanks for the issue.
